src/libs/MoneyRequestUtils.ts

```diff
@@ -42,8 +49,12 @@ function calculateAmountLength(amount: string): number {
 /**
  * Check if amount is a decimal up to 3 digits
  */
-function validateAmount(amount: string): boolean {
-    const decimalNumberRegex = new RegExp(/^\d+(,\d+)*(\.\d{0,2})?$/, 'i');
+function validateAmount(amount: string, decimals: number): boolean {
+    const regexString =
+        decimals === 0
+            ? `^\\d+(,\\d+)?$` // don't allow decimal point if decimals === 0
```

Comment:
Yeah, instead of matching anywhere in the string, it may be better to explicitly match it till the end 👍 (not sure if this would change anything because we already have validation to prevent multiple decimal dots).